python/equistore-operations/equistore/operations/join.py

```diff
@@ -218,13 +233,9 @@ def join(
     #
     #   [('a', 'b', 'c'), ('a', 'b')] -> ['a', 'b', 'c', 'a', 'b']
     #
-    # A nested list with sublist of different shapes can not be handled by np.unique.
-    unique_names = np.unique(functools.reduce(operator.concat, names_list))
-
-    # Label names are unique: We can do an equal check only checking the lengths.
-    names_are_same = np.all(
-        len(unique_names) == np.array([len(names) for names in names_list])
-    )
+    # A nested list with sublist of different shapes can not be handled by `set`.
+    unique_names = set(functools.reduce(operator.concat, names_list))
```

Comment:
I feel like this will not work with TorchScript. Could we rewrite this without using functools?

Reply:
Of course it doesn't work

```python
import functools
import operator

import torch

@torch.jit.script
def func() -> str:
    names_list = [["1", "2"], ["1", "5"]]
    return set(functools.reduce(operator.concat, names_list))

func()
```

throws an error. But, this is just a list of lists. We most likely can do it with a comprehension.